applications/ShapeOptimizationApplication/python_scripts/model_part_controller_factory.py

```diff
@@ -121,7 +146,18 @@ def GetDesignSurface(self):
         return self.design_surface
 
     # --------------------------------------------------------------------------
-    def DampNodalVariableIfSpecified(self, variable):
+    def DampNodalSensitivityVariableIfSpecified(self, variable):
+        if self.model_settings["damping"]["apply_damping"].GetBool():
+            self.damping_utility.DampNodalVariable(variable)
+
+        for direction_damping in reversed(self.direction_dampings):
+            direction_damping.DampNodalVariable(variable)
```

Comment:
So `damping_utility` takes into account all/multiple damping regions at the same time, while the new `direction_damping_utility` can't do that, because its damping operation has to be performed in reversed order?

Or is there another reason why `direction_damping_utility` can only deal with one model_part at a time?

Reply:
For standard damping this is possible, because it is sufficient to compute one scalar field of damping factors on the whole geometry. For damping of directions, it is necessary to know which damping factor is associated to which damping direction (there could be a different direction for each damping region).